app/src/main/java/me/guillaumin/android/osmtracker/db/TracklistAdapter.java

```diff
@@ -21,6 +23,34 @@
  */
 public class TracklistAdapter extends CursorAdapter {
 
+	public static String distanceToString(float distance, Resources resources) {
+		if (distance < 100)
+			return String.format("%d %s", Math.round(distance),
+					resources.getString(R.string.trackmgr_distance_meters));
+		else if (distance < 1000)
+			return String.format("%d %s", 10*Math.round(distance/10),
+					resources.getString(R.string.trackmgr_distance_meters));
+		else if (distance < 10000)
+			return String.format("%.1f %s", distance/1000,
+					resources.getString(R.string.trackmgr_distance_kilometers));
+		else
+			return String.format("%d %s", Math.round(distance/1000),
+					resources.getString(R.string.trackmgr_distance_kilometers));
+	}
+
+	public static String speedToString(float speed, Resources resources) {
+		float kmph = (float)3.6*speed;
+		if (kmph < 1)
+			return String.format("%.1g %s", kmph,
+					resources.getString(R.string.trackmgr_speed_kmph));
+		else if (kmph < 20)

```

Comment:
This is math, so not necessarily. Like for example, in that code copy-pasted from osmand, number of degrees in a flat angle (180) is used directly rather than as a named constant.